### PR TITLE
mm/kmm_heap : Support Kernel multi-heap

### PR DESCRIPTION
**Pull request by jeongchanKim**
There are two cases to use kernel multi-heap regions.
1. continuous physical RAM but want to split regions
2. separated physical RAMs but want to use for heap

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [447811c132ccf6dc3699fdbb1f5f20290dfd7a68] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [447811c132ccf6dc3699fdbb1f5f20290dfd7a68] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [0615944473fa084fa0e3ffa1bcbaf5acbe850750] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0615944473fa084fa0e3ffa1bcbaf5acbe850750] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [c9ad7b69808755c2e1ff2c85b3eeccbc6e002cad] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [c9ad7b69808755c2e1ff2c85b3eeccbc6e002cad] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3aded06a240dc854c5f187686b3c6e6e2ed79c84] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3aded06a240dc854c5f187686b3c6e6e2ed79c84] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3e4cdc22830036faa96e83e3c4380603d9f7c27e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3e4cdc22830036faa96e83e3c4380603d9f7c27e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a94f61ffa87bfeaac5c142e27c2c734fdf91cb01] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a94f61ffa87bfeaac5c142e27c2c734fdf91cb01] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [43150cca484fda33e4b38beaa1723d059944c900] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [43150cca484fda33e4b38beaa1723d059944c900] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [aa8c9d97ad17c5f38ffddb5dc77c0c8d271b9838] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [aa8c9d97ad17c5f38ffddb5dc77c0c8d271b9838] - Code Rule Check OK.

File: os/arch/arm/src/common/up_allocateheap.c [hunk 1]
```diff
@@ -209,10 +209,29 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
 }
 #endif
 
+/****************************************************************************
+ * Name: up_add_kregion
+ ****************************************************************************/
+#if defined(CONFIG_BUILD_PROTECTED) && (CONFIG_KMM_REGIONS > 1)
```
Comment (jeongarmy) [inline os/arch/arm/src/common/up_allocateheap.c]: Is it ok when CONFIG_BUILD_KERNEL?
I think it should be not flat build, isn't it?

File: os/arch/Kconfig [hunk 2]
```diff
@@ -798,6 +798,40 @@ config RAM_REGIONx_HEAP_INDEX
 		It means that its region can be handled to the index-th heap. 
 		Index can be from 0 to (MM_NHEAPS-1).
 
+config RAM_KREGIONx_START
+	string "An address or list of start address for kernel RAM region"
+	default "0x00000000,"
+	---help---
+		The address or address list of the RAM regions.
+		If you want to use more than two RAMs physically, or
```
Comment (jeongarmy) [inline os/arch/Kconfig]: 'more than 1' or 'more than equal to 2' is right.

File: os/mm/kmm_heap/kmm_addregion.c [hunk 3]
```diff
@@ -97,7 +97,8 @@
 
 void kmm_addregion(FAR void *heap_start, size_t heap_size)
 {
-	return mm_addregion(kmm_get_heap(), heap_start, heap_size);
+	struct mm_heap_s *kheap = kmm_get_heap();
+	return mm_addregion(&kheap[0], heap_start, heap_size);
```
Comment (jeongarmy) [inline os/mm/kmm_heap/kmm_addregion.c]: I think you can just put kheap or kmm_get_heap().

**Comment by seinfra** [comment]
Target : [b90283b0723812ad75521258ac9429fff361d47c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b90283b0723812ad75521258ac9429fff361d47c] - Code Rule Check (C++) OK.

**Pull request merged**